receiver/otlpreceiver/otlphttp.go

```diff
@@ -36,7 +38,20 @@ func handleTraces(resp http.ResponseWriter, req *http.Request, tracesReceiver *t
 
 	otlpResp, err := tracesReceiver.Export(req.Context(), otlpReq)
 	if err != nil {
-		writeError(resp, encoder, err, http.StatusInternalServerError)
+		httpStatus := http.StatusInternalServerError // the default status on errors
```

Comment:
Note to self: we want to return this error no matter what if we have a PermanentError.